Serenity Updates -  September 2020 - Options labels in product export via API (#750)
akeneo/pim-helpcenter: content/updates/2020-08/2-API-search.md

---
pim_announcement_img: ./img/api-filters.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

# New updates to ease API catalog structure synchronization
::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

> A **new `search` parameter** enables you to add filters to catalog structure endpoints. It will help reduce the number of data to process and the size of the API call responses. Moreover, we added **attribute group labels** inside the attribute endpoints to help you reduce the number of API calls needed to synchronize your catalog structure. 

Reviewer: ```suggestion
A **new `search` parameter** enables you to add filters when retrieving attributes. Moreover, we added **attribute group labels** inside the attribute endpoints.

Thos two enhancements will help reduce the number of data to process, the size of the API call responses, and the number of API calls needed to synchronize your catalog structure. 
```

## The `search` parameter  

Here are the available filters:

> - **updated date**: allows you to retrieve only data updated since a specific date and hour (available for attribute groups, attributes, and families), 

Reviewer: You'll need to reformulate this part as the search parameter is only available for attributes.

> - **updated date**: allows you to retrieve only data updated since a specific date and hour (available for attribute groups, attributes, and families), 

Reviewer: ```suggestion
- **updated date**: allows you to retrieve only data updated since a specific date and hour (available for attribute groups, and families), 
```

> - **code**: allows you to filter on specific codes (available for attribute groups, attributes, categories and families), 

Reviewer: ```suggestion
- **code**: allows you to filter on specific codes (available for attribute groups, categories and families), 
```

> - **type**: allows you to filter on attributes from a specific type (only available for attributes),
> - **parent**: allows you to retrieve all sub-categories of a parent category (only available for categories).  

Reviewer: ```suggestion
- **parent**: allows you to retrieve all sub-categories of a parent category (only available for categories).  
```

You can combine filters to achieve powerful results!

> Let's take an example: you have an eCommerce platform that uses your PIM data. You know the list of all the attributes you need, and your application synchronizes the catalog structure every day at 3 PM.  

Reviewer: Example to change for another.
